### PR TITLE
Deterministic data loader

### PR DESCRIPTION
**Pull request by palonso**
The goal is to have deterministic training so that two trails results in the same accuracy. 
This seed number is increased on each epoch so that a different offset is selected from each track.
Otherwise, the model would have the same view of the dataset on every epoch.

### CODE REVIEW

File: src/data_loaders.py [hunk 1]
```diff
@@ -49,8 +48,10 @@ def data_gen_standard(id, audio_repr_path, gt, pack):
         # let's deliver some data!
         if sampling == 'random':
             for i in range(0, param_sampling):
-                random_frame_offset = random.randint(
-                    0, frames_num - config['xInput'])
+                random_uniform = np.random.random()
```
Comment (Bomme) [inline src/data_loaders.py:55]: Why not use [numpy.random.randint](https://docs.scipy.org/doc/numpy-1.16.0/reference/generated/numpy.random.randint.html)?
(I think, to match the behaviour of `random.randint` you need to add `+1` to the upper bound)

File: src/train.py [hunk 2]
```diff
@@ -248,6 +248,11 @@ def model_and_cost(config, is_train):
 
     for i in range(config['epochs']):
         # training: do not train first epoch, to see random weights behaviour
+
+        # modify the seed number on every epoch so that we get different patches
+        # from each track
+        np.random.seed(seed=config['seed'] + i)
```
Comment (Bomme) [inline src/train.py:254]: are you sure this is necessary? Subsequent calls to a random number generator should give different results every time?

**Review by Bomme — COMMENTED** [review]
Nice change to use the numpy controlled random number generator! 

Maybe we can simplify this change a bit? (--> Less is more)

File: src/data_loaders.py (hunk 1 above)
Comment (palonso) [reply src/data_loaders.py:55]: There was a reason for this, but it was not evident. I added an explanatory comment.

File: src/train.py (hunk 2 above)
Comment (palonso) [reply src/train.py:254]: My intuition was exactly that, but empirically I found that the order was repeated on every epoch unless I added that line.
Now I'm looking into the problem again, and I believe that has to do with the parallel nature of the `Pescador` sampler. I found  [this issue](https://github.com/numpy/numpy/issues/9650#issuecomment-327139776) abut the problem.
My bet is that, as the sampling happens under a subthread for `Pesdacor`, it does not affect the `RandomState` of Numpy in the main thread. So unless the seed number is changed, in every epoch the sampler starts from the same point.

**Review by Bomme — APPROVED** [review]

**Pull request merged**